core/workspace.ts

```diff
@@ -804,6 +804,28 @@ export class Workspace implements IASTNodeLocation {
     return this.procedureMap;
   }
 
+  /**
+   * Returns the root workspace of this workspace if the workspace has
+   * parent(s).
+   *
+   * E.g. workspaces in flyouts and mini workspace bubbles have parent
+   * workspaces.
+   */
+  getRootWorkspace(): Workspace | null {
```

Comment:
Nit: might `getParentWorkspace` or `getSoureWorkspace` or `getOuterWorkspace` be a better name, since this is returns the next-enclosing workspace, not (as previously discussed) necessarily the root?

Reply:
`getParent` workspace doesn't quite work because for flyouts in a mini workspace bubble, their parent workspace is the workspace of the mini workspace bubble, but we actually return the workspace /containing/ the mini workspace bubble.

I think it's close enough to the truth to say "this always returns the root workspace" because further nesting than that is not supported. If we were to add support for further nesting (e.g. mini workspace bubbles inside eachother), updating this method would be part of that.